### ISSUE TITLE
Include warnings-count in report

### ISSUE THREAD

**Issue by tfry-git** [arduino/report-size-deltas#86]
### Describe the request

The compile-sketches actions allows the option enable-warnings-report in addition to enable-deltas-report.

Please include that info in the summary posted to PRs. In particular in a matrix build, involving different compilers, it is very useful to spot new (platform specific) warnings, quickly.

### Describe the current behavior

This action only posts a readily visible summary of the size-deltas, while - AFAICS - the count of warnings can only be found looking at the action logs itself, which is not very visible, and not very helpful, when dealing with a matrix build.

### 'arduino/report-size-deltas' version

1

### Issue checklist

- [X] I searched for previous requests in [the issue tracker](https://github.com/arduino/report-size-deltas/issues?q=)
- [X] I verified the feature was still missing when using the latest version
- [X] My request contains all necessary details

**Comment by tfry-git** [comment]
You are right, the name report-size-deltas would be off, if this feature was included. OTOH it would still serve a very similar purpose in the development workflow: Automated monitoring of the quality implications of a change / PR. Whatever the naming, it would be useful to a project maintainer to have all such information in one place.

This is the reasoning that led me to file my wish, here.

**Comment by 2bndy5** [comment]
FYI, the reports (json artifacts) that this actions needs from the compile-sketches action does not include the error/warning info you wish to be reported.

**Comment by tfry-git** [comment]
> FYI, the reports (json artifacts) that this actions needs from the compile-sketches action does not include the error/warning info you wish to be reported.

They do (if enable-warnings-report is enabled). Example artifact: https://github.com/sensorium/Mozzi/actions/runs/9160671044/artifacts/1519589989 . Excerpt for one sketch:

```
      "board": "arduino:avr:uno",
      "sketches": [
        {
          "name": "examples/01.Basics/Control_Gain",
          "compilation_success": true,
          "sizes": [
            {
              "name": "flash",
              ...
            },
            {
              "name": "RAM for global variables",
              ...
            }
          ],
          "warnings": {
            "current": {
              "absolute": 4
            },
            "previous": {
              "absolute": 7
            },
            "delta": {
              "absolute": -3
            }
          }
        },
        ...
      }
```
